plaidml/edsl/edsl_test.cc

```diff
@@ -46,6 +46,50 @@ Tensor Softmax(const Tensor& X) {
   return E / N;
 }
 
+TEST(CppEdsl, LowerPrecisionInvalidNegative) {
+  auto A = Placeholder(DType::FLOAT32, {3, 3});
+  auto C = A * (-2);
+
+  EXPECT_ANY_THROW({ Program("lower_precision", {C}, DType::FLOAT16, DType::UINT16); });
```

Comment:
I'm not sure why we shouldn't allow this case. I would assume that the result would be `f16` and that truncation would be happening somewhere in the middle of the program.

Reply:
`-2` is an int, which shouldn't be cast to `uint16`